### PR TITLE
Creation of Vocabs DB

### PR DESCRIPTION
**Pull request by kenyiu**

### CODE REVIEW

**Comment by bochecha** [comment]
So, first of all, I don't want to promise you a quick review or anything, because I think we should really focus on getting the new release of our stack out the door, before we start adding new features again.

That being said, thank you very much for starting work on this, it is a very desirable feature. :-)

A few quick thoughts:
1. This is not a part of libcangjie. :-)
   
   Please replace all mentions of "libcangjie" by "libchivocab". It's a temporary name, but it's still better than an incorrect name.
   
   Similarly, don't call the files `cangjie*`, the symbols `CANGJIE_*`, etc...
2. Don't commit binary/generated files to the VCS.
   
   As far as I can see, that goes for `src/vocabs_db` and `src/vocabs_dbbuilder`.
   
   Basically, if it's not a source file, it should not be in the VCS.
3. Make sure you include all the sources
   
   There is no source file for the `src/vocab_db`. Your `src/vocabs_dbbuilder.c` clearly tries parsing a table file, but you didn't add any.

---

As I said, I won't review further than that, because I want to focus on releasing libcangjie 1.0, pycangjie 1.0 and ibus-cangjie 2.0 as soon as possible.

I do believe that as it is right now, this is not suitable to be merged, though.

Please do go on working on this, though, including fixing the issues I pointed above.

We very much want to have such a feature at some point, so I'm very happy that you stepped up to implement it. :-)

**Comment by yookoala** [comment]
@bochecha This library is not target to be included in the up coming ibus-cangjie. It does not require you to promise that. Although it might wants to be in the next one, it will only be included after it is ready. Right now this is only preliminary codes.

So don't worry. Go ahead and finish your work on the current release.

**Comment by bochecha** [comment]
@yookoala pointed to me that the vocabs.txt file already exists. I had missed that.

@kenyiu, please ignore my point number 3 above. :-)

**Comment by mahiuchun** [comment]
it's 2015 guys :(

**Comment by mbridon** [comment]
Hi @kenyiu and @mahiuchun. I'm really sorry I never replied here.

I was very busy at first, and then I spent 2 years in a hospital starting back in 2020 and have been in re-education since then. :disappointed: 

I still very much want this feature in a future release of ibus-cangjie obviously though !

Do you think you might ever get motivated to resume working on this ? :confounded: 

If so, I promise I will try to support that work and review code in a timely manner. It's the least I can do now. :wink:
